client/asset/dcr/externaltx.go

```diff
@@ -328,13 +328,13 @@ func (dcr *ExchangeWallet) isOutputSpent(ctx context.Context, output *outputSpen
 	nextScanHeight := lastScannedHeight + 1
 
 	bestBlock := dcr.cachedBestBlock()
-	if nextScanHeight >= bestBlock.height {
+	if lastScannedHeight >= bestBlock.height {
 		// When a transaction is initially found in a block, that block is also
 		// scanned for spending transactions (see findTxInBlock) as an
 		// optimization since the block was pulled. As such, it is normal for
 		// the lastScannedHeight for newly located transactions to be the best
 		// block, so there is no need to warn, only return.
-		if nextScanHeight > bestBlock.height+1 {
+		if lastScannedHeight > bestBlock.height {
 			dcr.log.Warnf("Attempted to look for output spender in block %d but current tip is %d!",
 				nextScanHeight, bestBlock.height)
 		}
```

Comment:
This new logic looks right, although the pseudocode from Wisdom still reads easier to me even though there are separate `return`s

```go
if lastScannedHeight == bestBlock.height {
	// this is fine, just return, no more blocks to scan
	return false, nil
}
if lastScannedHeight > bestBlock.height {
	// not fine, how did it scan a height above the best height?
	// log a warning, should never happen anyways
	dcr.log.Warn("shouldnt happen asdf")
	// return too, since there're obviously no more blocks to scan
	return false, nil
}
```

Reply:
Will change to the more readable code here.